Detectors/TPC/simulation/include/TPCSimulation/DigitContainer.h

```diff
@@ -34,29 +34,20 @@ class DigitContainer{
   public:
     
     /// Default constructor
-    DigitContainer();
+    DigitContainer(int sectorID);
 
     /// Destructor
     ~DigitContainer() = default;
 
-    void reset();
-
-    /// Get the size of the container
-    /// \return Size of the CRU container
-    size_t getSize() const {return mCRU.size();}
-
-    /// Get the number of entries in the container
-    /// \return Number of entries in the CRU container
-    int getNentries() const;
-
     /// Add digit to the container
+    /// \param eventID MC Event ID
     /// \param hitID MC Hit ID
     /// \param cru CRU of the digit
     /// \param row Pad row of digit
     /// \param pad Pad of digit
     /// \param timeBin Time bin of the digit
     /// \param charge Charge of the digit
-    void addDigit(size_t hitID, int cru, int timeBin, int row, int pad, float charge);
+    void addDigit(int eventID, size_t hitID, CRU cru, int timeBin, GlobalPadNumber globalPad, float charge);
```

Comment:
Should {{CRU}} be modifiable?
Is a copy needed?
Otherwise consider {{const CRU& cru}}.
This is true also for other places where you introduced {{CRU}}.